### PR TITLE
SONARIAC-902 Rule S6380: Authorizing anonymous access to Azure resources is security-sensitive - `DataFactory/factories/linkedservices`

### PR DESCRIPTION
**Pull request by petertrr**

### CODE REVIEW

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/AnonymousAccessToResourceCheckTest.java [hunk 1]
```diff
@@ -36,8 +36,19 @@ void shouldFindIssuesInWebSitesResourceJson() {
       Verifier.issue(70, 10, 70, 40, "Make sure that disabling authentication is safe here."));
   }
 
+  @Test
+  void shouldFindIssuesInDataFactoryJson() {
+    ArmVerifier.verify("AnonymousAccessToResourceCheck/Microsoft.DataFactory_factories_linkedservices.json", check,
+      Verifier.issue(12, 10, 12, 43, "Make sure that authorizing anonymous access is safe here."));
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/AnonymousAccessToResourceCheckTest.java:42]: static imports

File: iac-extensions/arm/src/test/resources/checks/AnonymousAccessToResourceCheck/Microsoft.DataFactory_factories_linkedservices.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          "authenticationType": "Anonymous"
+        }
+      }
+    },
+    {
+      "type": "Microsoft.DataFactory/factories/linkedservices",
+      "apiVersion": "2018-06-01",
+      "name": "Compliant",
+      "properties": {
+        "type": "Web",
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/resources/checks/AnonymousAccessToResourceCheck/Microsoft.DataFactory_factories_linkedservices.json:21]: There should be a check for each type, see `ManagedIdentityCheckTest` for inspiration

File: iac-extensions/arm/src/test/resources/checks/AnonymousAccessToResourceCheck/Microsoft.DataFactory_factories_linkedservices.bicep [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            }
+        }
+    }
+}
+
+
+resource linkedService 'Microsoft.DataFactory/factories/linkedservices@2018-06-01' = {
+    name: 'example'
+    properties: {
+            // Compliant - not a sensitive type
```
Comment (mstachniuk) [inline iac-extensions/arm/src/test/resources/checks/AnonymousAccessToResourceCheck/Microsoft.DataFactory_factories_linkedservices.bicep:35]: Nitpick - indentation

**Review by mstachniuk — APPROVED** [review]
LGTM and see my comments

File: iac-extensions/arm/src/test/resources/checks/AnonymousAccessToResourceCheck/Microsoft.DataFactory_factories_linkedservices.json (hunk 2 above)
Comment (petertrr) [reply iac-extensions/arm/src/test/resources/checks/AnonymousAccessToResourceCheck/Microsoft.DataFactory_factories_linkedservices.json]: Implemented for JSON; still we don't yet support `Noncompliant` comment in `BicepVerifier#verifyContent`. Probably, we should add it in the future

**Pull request merged**